crates/router/src/core/payments/flows/session_flow.rs

```diff
@@ -253,6 +286,72 @@ async fn create_applepay_session_token(
     }
 }
 
+fn get_session_request_for_simplified_apple_pay(
+    session_token_data: payment_types::SessionTokenForSimplifiedApplePay,
+) -> RouterResult<payment_types::ApplepaySessionRequest> {
+    let request = payment_types::ApplepaySessionRequest {
+        merchant_identifier: consts::APPLE_PAY_MERCHANT_ID.to_owned(),
+        display_name: "Apple pay".to_string(),
+        initiative: "web".to_string(),
+        initiative_context: session_token_data.initiative_context,
+    };
+    Ok(request)
+}
+
+fn get_session_request_for_manual_apple_pay(
+    session_token_data: payment_types::SessionTokenInfo,
+) -> RouterResult<payment_types::ApplepaySessionRequest> {
```

Comment:
```suggestion
) -> payment_types::ApplepaySessionRequest {
```
Return the without warping inside Result